pages/instructorAssessmentGroups/instructorAssessmentGroups.js

```diff
@@ -146,67 +158,112 @@ router.post('/', function(req, res, next) {
             if (ERR(err, next)) return;
             res.redirect(req.originalUrl);
         });
-    } else if (req.body.__action == 'addmember') {
+    } else if (req.body.__action == 'add_member') {
         const assessment_id = res.locals.assessment.id;
         const gid = req.body.gid;
-        const uids = req.body.addmemberuids;
+        const uids = req.body.add_member_uids;
         const uidlist = uids.split(/[ ,]+/);
         let failedUids = '';
         res.locals.errormsg = '';
-        (async () => {
-            for (const uid of uidlist) {
-                const params = [
-                    assessment_id,
-                    gid,
-                    uid,
-                    res.locals.authn_user.user_id,
-                ];
-                try {
-                    await sqldb.callAsync('assessment_groups_add_member', params);
-                } catch (err) {
-                    failedUids += '[' + uid + '] ';
-                }
-            }
-            if (failedUids.length > 0) {
-                res.locals.errormsg += 'Failed to add ' + failedUids + 'to Group No.' + gid + '. Please check if the uid exist.\n';
+
+        //verify the group belong to the assessment first
+        let params = {
+            assessment_id,
+            gid,
```

Comment:
In PL we use `TABLENAME_id` to refer to the `id` column in tables. So I think this should be `group_id`. Names like `qid` for questions refer to the string name of the question, not the `question_id`. Sorry! I know this is confusing.